gen3/bin/dbbackup.sh

```diff
@@ -222,28 +224,289 @@ copy_to_aurora() {
+- apiGroups: ["batch"]
+  resources: ["jobs", "cronjobs"]
+  verbs: ["get", "watch", "list", "create", "delete", "patch", "update"]
+- apiGroups: ["apps"]
+  resources: ["deployments", "statefulsets", "daemonsets"]
+  verbs: ["get", "watch", "list", "create", "delete", "patch", "update"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: dbencrypt-rolebinding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: dbencrypt-role
+subjects:
+- kind: ServiceAccount
+  name: dbencrypt-sa
+  namespace: ${namespace}
+EOF
+  fi
+
+# Install CSI driver
+gen3_log_info "eks cluster name: $eks_cluster"
+
+# Capture the output of the command and prevent it from exiting the script
+csi_driver_check=$(aws eks describe-addon --cluster-name $eks_cluster --addon-name aws-mountpoint-s3-csi-driver --query 'addon.addonName' --output text 2>&1 || true)
+
+if echo "$csi_driver_check" | grep -q "ResourceNotFoundException"; then
+  gen3_log_info "CSI driver not found, installing..."
+  aws eks create-addon --cluster-name $eks_cluster --addon-name aws-mountpoint-s3-csi-driver --service-account-role-arn arn:aws:iam::${account_id}:role/AmazonEKS_S3_CSI_DriverRole
+  sleep 20
+  # Check CSI driver installation status
+  csi_status=$(aws eks describe-addon --cluster-name $eks_cluster --addon-name aws-mountpoint-s3-csi-driver --query 'addon.status' --output text)
+  if [ "$csi_status" == "ACTIVE" ]; then
+    gen3_log_info "CSI driver successfully installed and active."
+  else
+    gen3_log_info "CSI driver installation failed or not active. Current status: $csi_status"
+    exit 1
+  fi
+elif echo "$csi_driver_check" | grep -q "aws-mountpoint-s3-csi-driver"; then
+  gen3_log_info "CSI driver already exists, skipping installation."
+else
+  gen3_log_info "Unexpected error occurred: $csi_driver_check"
+  exit 1
+fi
+
+  if ! kubectl get pv s3-pv-db-backups 2>&1; then
+    cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: s3-pv-db-backups
+spec:
+  capacity:
+    storage: 120Gi
+  accessModes:
+    - ReadWriteMany
+  mountOptions:
+    - allow-delete
+    - allow-other
+    - uid=1000
+    - gid=1000
+    - region us-east-1
+    - sse aws:kms
+    - sse-kms-key-id ${kms_key_arn}
+  csi:
+    driver: s3.csi.aws.com
+    volumeHandle: s3-csi-db-backups-volume
+    volumeAttributes:
+      bucketName: ${bucket_name_encrypted}
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: s3-pvc-db-backups
+spec:
+  accessModes:
+    - ReadWriteMany
+  storageClassName: ""
+  resources:
+    requests:
+      storage: 120Gi
+  volumeName: s3-pv-db-backups
+EOF
+  fi
+}
+
+# Check prerequisites for encrypted backup and cronjob
+check_prerequisites() {
+    create_or_get_kms_key
+    create_s3_bucket $bucket_name $kms_key_arn
+    create_s3_bucket $bucket_name_encrypted $kms_key_arn
+    setup_csi_driver
```

Comment:
Ideally this should be a separate script or terraform. Then we can call that separate script or terraform from within this script.